index_integration_test.go

```diff
@@ -83,6 +83,23 @@ func TestIndex_CursorDynamics(t *testing.T) {
 		assert.Equal(t, 2, found)
 	})
 
+	t.Run("2 docs found on single prefix key using duplicate key", func(t *testing.T) {
+		q := New(Prefix(key1, MustParseScalar("1"))).And(
```

Comment:
What would happen if you'd supply the following:

```go
		q := New(Prefix(key1, MustParseScalar("1"))).And(
			Prefix(key1, MustParseScalar("2"))).
			Or(Eq(key1, , MustParseScalar("3")))
```

(meaning: `key == 1 && key == 2 || key == 3`)